### PR TITLE
Consistent Extrudability for FillBetweenPointVectorsTools

### LINKED ISSUE #21141: Consistent Extrudability for FillBetweenPointVectorsTools
## Bug Description
Depending on the input Point vectors, the elements in the 2D mesh in XY plane created by `FillBetweenPointVectorsTools` may either be extruded in '0 0 1' direction or '0 0 -1' direction. Wrong extrusion direction leads to negative element volume and thus negative Jacobian. Here, a short-term solution would be to make sure that `FillBetweenPointVectorsTools` consistently produces meshes that can be extruded in '0 0 1' direction.

## Steps to Reproduce
In `FillBetweenPointVectorsGenerator`, flipping or not flipping both `positions_vector_1` and `positions_vector_2` will produce two meshes. One can be extruded in '0 0 1' direction, the other can be extruded in '0 0 -1' direction.

## Impact
Facilitate the creation of 3D reactor meshes.

### PR DESCRIPTION
**Pull request by miaoyinb**
closes #21141

### CODE REVIEW

**Comment by miaoyinb** [comment]
@GiudGiud. A quick fix for `FillBetweenPointVectorsTools`.

As I mentioned in  #21141, more work might need to be done in `FancyExtruderGenerator` or separately to improve consistence in extrusion in the future.

File: framework/src/utils/FillBetweenPointVectorsTools.C [hunk 1]
```diff
@@ -292,56 +309,117 @@ elementsCreationFromNodesVectors(ReplicatedMesh & mesh,
...
+        elem->set_node(1) = nodes[i][nodes_down_it];
+        elem->set_node(2) = nodes[i][nodes_down_it + 1];
+      }
+      else
+      {
+        elem->set_node(0) = nodes[i + 1][nodes_up_it];
+        elem->set_node(2) = nodes[i][nodes_down_it];
+        elem->set_node(1) = nodes[i][nodes_down_it + 1];
+        is_elem_flip = true;
+      }
```
Comment (GiudGiud) [inline framework/src/utils/FillBetweenPointVectorsTools.C:416]: since you have 3 times the same code we could have a routine in the tools?

**Review by GiudGiud — COMMENTED** [review]
exodiffs are only element renumbering right?

File: framework/src/utils/FillBetweenPointVectorsTools.C (hunk 1 above)
Comment (miaoyinb) [reply framework/src/utils/FillBetweenPointVectorsTools.C:416]: Yes, you are right

File: framework/include/utils/FillBetweenPointVectorsTools.h [hunk 2]
```diff
@@ -245,4 +245,36 @@ bool isBoundaryOpenSingleSegment(ReplicatedMesh & mesh,
...
+
+/**
+ * Creates an QUAD4 element that can be extruded in (0 0 1) direction.
+ * @param elem pointer of the element to be created
+ * @param nd_0 pointer of element's first node
+ * @param nd_1 pointer of element's second node
+ * @param nd_2 pointer of element's third node
+ * @param nd_3 pointer of element's fourth node
+ * @param transition_layer_id id of the subdomain that this element belongs to
+ * @return whether the element is flipped to ensure (0 0 1) extrudability
```
Comment (GiudGiud) [inline framework/include/utils/FillBetweenPointVectorsTools.h:257]: so are they not (0 0 -1) extrudable then?

Comment (miaoyinb) [reply framework/include/utils/FillBetweenPointVectorsTools.h:257]: In the current version of `FancyExtruderGenerator`, if an element can be extruded in (0 0 1) direction, it cannot be extruded in (0 0 -1) direction (negative Jacobian/volume due to the wrong node order)

Comment (GiudGiud) [reply framework/include/utils/FillBetweenPointVectorsTools.h:257]: is there an issue about this?

**Review by GiudGiud — APPROVED** [review]

File: framework/include/utils/FillBetweenPointVectorsTools.h (hunk 2 above)
Comment (miaoyinb) [reply framework/include/utils/FillBetweenPointVectorsTools.h:257]: > is there an issue about this?

Thanks for your review. I will create an issue for this later.

**Pull request merged**